.github/workflows/nightly.yml

```diff
@@ -45,3 +51,59 @@ jobs:
 
     - name: Run tests under valgrind
       run: ${{github.workspace}}/test/test_valgrind.sh ${{github.workspace}} ${{github.workspace}}/build ${{matrix.tool}}
+
+  coverity:
+    name: Coverity
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+
+    - name: Install apt packages
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y cmake hwloc libhwloc-dev libjemalloc-dev libnuma-dev libtbb-dev
+
+    - name: Download Coverity
+      run: |
+        wget -nv https://scan.coverity.com/download/linux64 --post-data "token=${{ secrets.COVERITY_SCAN_TOKEN }}&project=oneapi-src%2Funified-memory-framework" -O coverity_tool.tgz
+
+    - name: Extract Coverity
+      run: |
+        tar xzf coverity_tool.tgz
+
+    - name: Configure CMake
+      run: >
+        cmake
+        -B ${{github.workspace}}/build
+        -DCMAKE_BUILD_TYPE=Release
+        -DUMF_FORMAT_CODE_STYLE=OFF
+        -DUMF_DEVELOPER_MODE=OFF
+        -DUMF_ENABLE_POOL_TRACKING=OFF
+        -DUMF_BUILD_LIBUMF_POOL_SCALABLE=ON
+        -DUMF_BUILD_LIBUMF_POOL_DISJOINT=ON
+        -DUMF_BUILD_LIBUMF_POOL_JEMALLOC=ON
+
+    - name: Build
+      run: |
+        export COVERITY_DIR=$(find . -maxdepth 1 -type d -name "cov-analysis-linux64-*" | head -n 1)
+        if [ -n "$COVERITY_DIR" ]; then
+          export PATH="$PATH:$COVERITY_DIR/bin"
```

Comment:
hmm... does this work? is this dir required in this step only or also in some of the next steps?

I believe setting env. vars in GHA is (was?) a little different (pls see e.g. [here](https://stackoverflow.com/questions/68211182/how-to-set-system-path-variable-in-github-action-workflow) or [here](https://stackoverflow.com/a/57969570))

If this works as expected, then pls disregard my comment 😉